tests/phpunit/api/v3/ACLPermissionTest.php

```diff
@@ -95,7 +95,21 @@ public function testContactGetNoResultsHookWithEditMyContact() {
       'check_permissions' => 1,
       'return' => 'display_name',
     ));
-    $this->assertEquals(0, $result['count']);
+    $this->assertEquals(1, $result['count']);
+  }
+
+  /**
+   * Function tests that a user with "edit my contact" can edit themselves.
+   */
+  public function testContactEditHookWithEditMyContact() {
+    $this->markTestIncomplete('api acls only work with contact get so far');
+    $cid = $this->createLoggedInUser();
+    $this->hookClass->setHook('civicrm_aclWhereClause', array($this, 'aclWhereHookNoResults'));
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = array('access CiviCRM', 'edit my contact');
+    $this->callAPISuccess('contact', 'create', array(
+      'check_permissions' => 1,
+      'id' => $cid,
+    ));
```

Comment:
@eileenmcnaughton I wrote this test but then realized the api doesn't respect acls for contact create or update so marked as incomplete. What do you think it would take to get it working tho?